drivers/misc/nxp_lp_flexcomm/nxp_lp_flexcomm.c

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright 2024 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT nxp_lp_flexcomm
+
+#include <errno.h>
+#include <zephyr/device.h>
+#include <zephyr/irq.h>
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+#include "nxp_lp_flexcomm.h"
+
+LOG_MODULE_REGISTER(nxp_lp_flexcomm, LOG_LEVEL_ERR);
+
+struct nxp_lp_flexcomm_child {
+	const struct device *dev;
+	uint8_t periph;
+	child_isr_t lp_flexcomm_child_isr;
```

Comment:
I will admit that registering the isr at runtime is something I did with the ENET driver since I also had to deal with a parent driver to handle IRQs, but I want to consider an approach of defining a linker section for a table of irq handlers for these parent drivers, if you dont want to deal with it in this PR then it's fine, but, combining this with the other comments I have on this struct, could remove all the RAM usage of this parent driver.